adi/lidar.py

```diff
@@ -0,0 +1,110 @@
+#       from this software without specific prior written permission.
+#     - The use of this software may or may not infringe the patent rights
+#       of one or more patent holders.  This license does not release you
+#       from the requirement that you obtain separate licenses from these
+#       patent holders to use this software.
+#     - Use of the software either in source or binary form, must be run
+#       on or directly connected to an Analog Devices Inc. component.
+#
+# THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+# INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT, MERCHANTABILITY AND FITNESS FOR A
+# PARTICULAR PURPOSE ARE DISCLAIMED.
+#
+# IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, INTELLECTUAL PROPERTY
+# RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR
+# BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+# STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+# THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from adi.ad5627 import ad5627
+from adi.ad9094 import ad9094
+from adi.rx_tx import phy
+import numpy as np
+
+class Lidar (ad5627, ad9094, phy):
+    """ LiDAR """
+    _device_name = "LiDAR"
+
+    def __init__(self, uri, pulse_capture_address="7c700000"):
+        ad5627.__init__(self, uri)
+        ad9094.__init__(self, uri)
+        # The name of the pulse capture channel. The address might be different,
+        # depending on how the board was built.
+        self.pulse_capture = pulse_capture_address + "." + "axi-pulse-capture"
+        self._ctrl = self._ctx.find_device(self.pulse_capture)
+        self.set_all_iio_attrs_to_default_values()
+
+    def read_channels(self):
+        """Read the buffers for all the enabled channels, except Channel4 which should
+        be all zeroes and not relevant for the user.
+        """
+        ret = []       
+        for channel in self.rx():
+            cast_channel = channel.astype(np.int8)
+            # Ignore Channel4 and all everything else.
+            if not np.all(cast_channel==0):
+                ret.append(cast_channel)
+        return ret
+
+    def laser_enable(self):
+        self._set_iio_attr_int("altvoltage0", "en", True, 1, self._ctrl)
+
+    def laser_disable(self):
+        self._set_iio_attr_int("altvoltage0", "en", True, 0, self._ctrl)
+
+    def set_laser_pulse_width(self, width):
+        self._set_iio_attr_int("altvoltage0", "pulse_width_ns", True, width, self._ctrl)
+
+    def set_laser_frequency(self, frequency):
+        self._set_iio_attr_int("altvoltage0", "frequency", True, frequency, self._ctrl)
+
+    def set_channel_sequencer_enable_disable(self, status):
+        """Status = 0 to disable, 1 to enable the sequencer."""
```

Comment:
Most other methods that address individual IIO properties are using decorators. Is there a specific reason this is not being done here as well?

Reply:
I'm not interested in reading this attribute, only in setting it.

Reply:
Would anyone ever want to know the state of this attribute? Also, what about the other set methods like: set_channel_sequencer_opmode, set_channel_sequencer_order_auto_mode, ...

Reply:
I've changed all the `set` interfaces to properties.